js/controller.js

```diff
@@ -1040,6 +1042,8 @@ OO.plugin('Html5Skin', function(OO, _, $, W) {
         this.state.screenToShow = CONSTANTS.SCREEN.SHARE_SCREEN;
       } else {
         this.state.screenToShow = CONSTANTS.SCREEN.END_SCREEN;
+        //TODO: END_SCREEN_SHOWN is not consumed by anyone, should we remove? If not, should
+        //we publish this when the end screen is actually shown?
```

Comment:
We probably should move the publish to after the renderskin. Also, I'd rather be thorough and send the message without anyone listening then having to figure out where to add this line later if we do need it.

Reply:
The render skin is not instant and is async. It would be a bigger change than moving the publish after the ```renderSkin``` call. I think this is outside the scope of this PR and was just something I noticed. Can we make this change in some future PR?

Reply:
Sure